File: dbLoader/Models/AuctionItem.cs
using DataHandler.Models;
using System.Text.Json;
using System.Text.Json.Serialization;
using Newtonsoft.Json;

namespace dbLoader.Models
{
    public class AuctionItem : IIdentifyable
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public List<string> PicturesPaths { get; set; } = new List<string>();
        public string AuctionId { get; set; } = "";
        public int StartingPrice { get; set; }
        public string SerialNumber { get; set; } = "";

        // replaced dictionary with single string to match spec 4.1
        [JsonProperty("additionalParameters")]
        public string AdditionalParameters { get; set; }
        
    // state of the item: e.g. "new", "used", "working", "notWorking"
    public string State { get; set; } = "new";

        public void PrintSelf()
        {
            Console.WriteLine($"Id : {this.Id}");
            Console.WriteLine($"Name : {this.Name}");
            Console.WriteLine($"Starting price: {this.StartingPrice}");
            foreach (var item in this.PicturesPaths)
            {
                Console.WriteLine($"picture: {item}");
            }
            Console.WriteLine($"Additional params: {this.AdditionalParameters}");
            Console.WriteLine($"State: {this.State}");
            Console.WriteLine($"Serial Number: {this.SerialNumber}");
        }
    }
}

File: EvidenAuctionHouse/src/models/auctionItem.ts
export interface AuctionItemOptions {
    id: string;
    name: string;
    startingPrice: number;
    additionalParameters?: string;
    auctionId?: string;
    serialNumber?: string;
    picturesPaths?: string[];
    state?: string;
}

export class AuctionItem {
    id: string;
    name: string;
    picturesPaths: string[] = [];
    auctionId: string = "";
    startingPrice: number;
    additionalParameters?: string;
    state?: string = "new";
    serialNumber?: string;

    constructor(opts: AuctionItemOptions) {
        this.id = opts.id;
        this.name = opts.name;
        this.startingPrice = opts.startingPrice;
        this.additionalParameters = opts.additionalParameters;
        this.auctionId = opts.auctionId ?? "";
        this.serialNumber = opts.serialNumber ?? "";
        this.picturesPaths = opts.picturesPaths ?? [];
        this.state = opts.state ?? "new";
    }
}